ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/fillers/CoreProblemFiller.java

```diff
@@ -26,79 +27,163 @@ public CoreProblemFiller(LinearRaoProblem linearRaoProblem, LinearRaoData linear
 
     @Override
     public void fill() {
-        Crac crac = linearRaoData.getCrac();
-        if (crac.getPreventiveState() != null) {
-            Set<RangeAction> rangeActions = crac.getRangeActions(linearRaoData.getNetwork(), crac.getPreventiveState(), UsageMethod.AVAILABLE);
-            rangeActions.forEach(this::fillRangeAction);
-            crac.getCnecs().forEach(cnec -> {
-                fillCnec(cnec);
-                rangeActions.forEach(rangeAction -> updateCnecConstraintWithRangeAction(cnec, rangeAction));
-            });
-        }
+        // add variables
+        buildFlowVariables();
+        buildRangeActionSetPointVariables();
+        buildRangeActionAbsoluteVariationVariables();
+
+        // add constraints
+        buildFlowConstraints();
+        buildRangeActionConstraints();
     }
 
     @Override
-    public void update(List<String> activatedRangeActionIds) {
-        Crac crac = linearRaoData.getCrac();
-        if (crac.getPreventiveState() != null) {
-            Set<RangeAction> rangeActions = crac.getRangeActions(linearRaoData.getNetwork(), crac.getPreventiveState(), UsageMethod.AVAILABLE);
-            activatedRangeActionIds.forEach(this::updateRangeActionBounds);
-            crac.getCnecs().forEach(cnec -> {
-                linearRaoProblem.updateReferenceFlow(cnec.getId(), linearRaoData.getReferenceFlow(cnec));
-                rangeActions.forEach(rangeAction -> updateCnecConstraintWithRangeAction(cnec, rangeAction));
-            });
-        }
+    public void update() {
+        // update reference flow and sensitivities of flow constraints
+        updateFlowConstraints();
```

Comment:
Not sure this is useful but ok

Reply:
the idea of the update method is : 
* no to rebuild from scratch an optimisation problem in which we just need to change a few values
* be able to benefit from hot-start in the solving of the optimisation problem, based on the previous optimal solution calculated.